miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultInstrumentService.java

```diff
@@ -0,0 +1,189 @@
+import java.util.Map;
+import java.util.function.Consumer;
+
+import org.hibernate.exception.ConstraintViolationException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import uk.ac.bbsrc.tgac.miso.core.data.Instrument;
+import uk.ac.bbsrc.tgac.miso.core.data.Run;
+import uk.ac.bbsrc.tgac.miso.core.data.ServiceRecord;
+import uk.ac.bbsrc.tgac.miso.core.data.type.PlatformType;
+import uk.ac.bbsrc.tgac.miso.core.store.InstrumentStore;
+import uk.ac.bbsrc.tgac.miso.core.store.ServiceRecordStore;
+import uk.ac.bbsrc.tgac.miso.core.util.PaginationFilter;
+import uk.ac.bbsrc.tgac.miso.service.InstrumentService;
+import uk.ac.bbsrc.tgac.miso.service.security.AuthorizationManager;
+
+@Transactional(rollbackFor = Exception.class)
+@Service
+public class DefaultInstrumentService implements InstrumentService {
+
+  @Autowired
+  private AuthorizationManager authorizationManager;
+  @Autowired
+  private InstrumentStore instrumentDao;
+  @Autowired
+  private ServiceRecordStore serviceRecordDao;
+  @Autowired
+  private RunService runService;
+
+  @Override
+  public Collection<Instrument> list() throws IOException {
+    return instrumentDao.listAll();
+  }
+
+  @Override
+  public Collection<Instrument> listByPlatformType(PlatformType platformType) throws IOException {
+    return instrumentDao.list(0, 0, true, "id", PaginationFilter.platformType(platformType), PaginationFilter.archived(false));
+  }
+
+  @Override
+  public Collection<ServiceRecord> listServiceRecordsByInstrument(long instrumentId) throws IOException {
+    return serviceRecordDao.listByInstrumentId(instrumentId);
+  }
+
+  @Override
+  public Instrument get(long instrumentId) throws IOException {
+    return instrumentDao.get(instrumentId);
+  }
+
+  @Override
+  public Instrument getByName(String name) throws IOException {
+    return instrumentDao.getByName(name);
+  }
+
+  @Override
+  public Instrument getByUpgradedInstrumentId(long instrumentId) throws IOException {
+    return instrumentDao.getByUpgradedInstrument(instrumentId);
+  }
+
+  @Override
+  public ServiceRecord getServiceRecord(long recordId) throws IOException {
+    return serviceRecordDao.get(recordId);
+  }
+
+  @Override
+  public Long create(Instrument instrument) throws IOException {
+    authorizationManager.throwIfNonAdmin();
+    loadChildEntities(instrument);
+    return save(instrument);
+  }
+
+  @Override
+  public void update(Instrument instrument) throws IOException {
+    authorizationManager.throwIfNonAdmin();
+    Instrument managed = get(instrument.getId());
+    applyChanges(managed, instrument);
+    loadChildEntities(managed);
+    save(managed);
+  }
+
+  private long save(Instrument instrument) throws IOException {
+    return instrumentDao.save(instrument);
+  }
+
+  private void applyChanges(Instrument target, Instrument source) throws IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)